### DISCUSSION TITLE
Is the recent RFC a mistake? v19 should be buried?

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#3424] [General]
Even if that RFC has been voted and is implemented already, please express your opinion about current OpenMage, about the decisions that are taken, Fabrizio's work, so that Sven has the opportunity to criticize with arguments in one place. 

Thank you.

**Comment by sreichel** [comment]
I am definitely happy with Fabrizios work. He (and others) kept up progress for last month. But sometimes we move to fast IMHO ...

(I will certainly answer in more detail later)

**Comment by addison74** [comment]
I still consider myself a person who wants to express himself and evolve. During the years spent here I felt wonderful with people with whom I have a common goal. I had contradictory discussions, dissatisfactions, I respected opinions, it was an unforgettable experience to see a gathering of souls from all over the globe, each with their uniqueness.

Even if Magento 1 was thrown into the trash in 2020, OpenMage remains its most valuable descendant, it is alive and well thanks to a hand of people. In addition, it is functional and can quickly set up an online business for anyone interested.

The RFC was discussed and voted on. Democratically, the result is respected, no matter if it displeases someone.

v19 had to be retired at some point and this was set for 2025. It was already very difficult to go in parallel with two branches. Everything that was proposed as PRs was analyzed before and it was decided in which branch it goes, practically v19 benefits from all contributions provided it does not affect its functionality. Anyone who wants can easily take parts of the code from v20 with git and implement them.

I advise those using v19 to start testing and moving to v20. DDEV is an excellent test environment Those who complain about this aspect should not forget that the extensions they use are not maintained, that PHP grows as versions. Any store that sells online should have a maintenance budget. Bringing extensions up to date is an investment, OpenMage ensures backward compatibility for nothing if it is not also for extensions and custom code. As well as updating the servers, recent Debian 12 comes with PHP 8.2.

For some time the maintainer has been Fabrizio and I want to thank him for his work. Without a locomotive we would have closed the shutters. The number of reported issues decreased by 70% compared to last year, we have less than 100 PRs to solve, we still have 25 PRs that we must not forget and an ideas section in Discussions. You only need to allocate a little time and continue working together as a team.

OpenMage is moving forward as long as we exist and we want to contribute. When the project can no longer be maintained, we shake hands virtually, we archive it and that's about it.

**Comment by elidrissidev** [comment]
First of all, a thank you to anyone who is/was keeping this project alive, and a special one to @fballiano for all his work (sometimes I wonder if his day is 28 hours long 😂).

Regarding the v19 thing, it was quite the work to maintain two versions and always having to account for which change is merged where, hence why the RFC was born to simplify it and allow the project to move forward while still maintaining some support to v19 till 2025 (albeit only security patches).  Some people seem to still want an LTS version, but unless the project gets financial support or they themselves start to contribute, I can't see it working too well for the few of us here.

**Comment by loekvangool** [comment]
I'm on v19 and I support the RFC. I love you all.

**Comment by colinmollenhour** [comment]
I've been away for a while but from the discussions I've seen so far, my take-aways are:

@Hanmac As @Flyingmana stated, you should submit a separate PR after the first is accepted into main which cherry-picks the accepted PR commit into v19 - the maintainers will do this for severe/security updates when they are critical but we expect the community to do it for non-critical updates due to limited resources. This could have been stated more clearly in the RFC but in short, it is discouraged and we strongly suggest just getting onto the latest version. There really isn't anything that should require a heavy lift to fix, it should just be a matter of running through some thorough testing before deploying (like any other Magento upgrade - probably easier than most).

To the point made by @sreichel in German regarding upgrade path from Magento to OpenMage, we probably need a more concise list of breaking changes (excluding new features that don't cause BC breaks) between Magento 1.9.4.5 and the latest OpenMage to make new migrations easier or at least less concerning about the unknowns. This is covered in the README.md but it is a bit scattered and could certainly be more concise since I think the list of true breaking changes is quite minimal. Any volunteers?

The threads containing heated disagreements on the v20.1.0 release discussion should be **deleted** as they don't belong in such prominent places as on release notes. I will delete them soon if there is no objection - not to quash opinions, just to tidy our public image as the discussion has been rightfully relocated to here.

@sreichel Please try to state your opinions and dissatisfactions less abrasively and in the appropriate channels - this is not a good look for the community and depletes the good will of all contributors and I agree with @Flyingmana that at some point we have to draw the line about what is accepted and what is not - ideally this is just something that is understood and respected by everyone so it doesn't have to be written anywhere or discussed further.

**Reply by colinmollenhour** [reply]
I've never deleted a comment before.. this must be what it would be like to work at Facebook.. 🤮 
I thought deleting a top-level comment would delete the comments below it but it doesn't.. So I just hid some of them. The one I deleted was "I'm just sad that my commits aren't backported into 19.x". - not delete-worthy but was thinking it would delete the rest of the discussion.

I hope we can call this water under the bridge and move forward with continuing the great work everyone is doing. 👍

**Comment by sreichel** [comment]
v19 is burried. 

(_imho_ ... we never ever had the manpower to maintain 3 branches - 19, main, next)
